### PR TITLE
New package: bash-language-server-4.9.1

### PR DESCRIPTION
**Pull request by sirkhancision**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc
- I built this PR locally for these architectures:
  - i686
  - x86_64-musl
  - aarch64

### CODE REVIEW

**Comment by sirkhancision** [comment]
This also resolves #38606

**Comment by icp1994** [comment]
Unfortunate to have a language server for bash being available only under `i686` & `x86_64`.

**Comment by classabbyamp** [comment]
how did you build this for x86_64-musl when `archs="i686 x86_64"`?

File: srcpkgs/bash-language-server/template [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	yarn run compile
+}
+
+do_install() {
+	TARGET_PATH="usr/lib/${pkgname}"
+	cd server
+
+	cp package.json{,.bak}
+	read -ra devDependencies < <(jq -r '.devDependencies | keys | join(" ")' package.json)
+	yarn remove --frozen-lockfile "${devDependencies[@]}"
```
Comment (classabbyamp) [inline srcpkgs/bash-language-server/template:26]:
```suggestion
	read -ra _devDependencies < <(jq -r '.devDependencies | keys | join(" ")' package.json)
	yarn remove --frozen-lockfile "${_devDependencies[@]}"
```

File: srcpkgs/bash-language-server/template [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	mv package.json{.bak,}
+
+	vmkdir ${TARGET_PATH}
+	vcopy * ${TARGET_PATH}
+
+	vmkdir usr/bin
+	ln -sf /${TARGET_PATH}/out/cli.js ${DESTDIR}/usr/bin/${pkgname}
+	chmod 0755 ${DESTDIR}/${TARGET_PATH}/out/cli.js
+
+	vlicense ${wrksrc}/LICENSE
```
Comment (classabbyamp) [inline srcpkgs/bash-language-server/template:36]:
```suggestion
	vlicense LICENSE
```

**Comment by classabbyamp** [comment]
> Unfortunate to have a language server for bash being available only under `i686` & `x86_64`.

cross-compiles fine for me 🤷‍♀️

**Comment by sirkhancision** [comment]
> how did you build this for x86_64-musl when `archs="i686 x86_64"`?

My bad, I had built it before restricting the architectures, and I thought x86_64 meant both glibc and musl. It's corrected now, and limiting i686 to glibc, as building it for me fails in other architectures.

**Comment by classabbyamp** [comment]
i686-musl doesn't really exist, so don't worry about it.

`[archname]` means glibc, `[archname]-musl` means musl, `[archname]-*` is both

File: srcpkgs/bash-language-server/template (hunk 2 above)
Comment (sirkhancision) [reply srcpkgs/bash-language-server/template:36]: This is because of the previous `cd` used, meaning that the license is at the child directory.

Comment (classabbyamp) [reply srcpkgs/bash-language-server/template:36]: ../LICENSE then

**Comment by sirkhancision** [comment]
> i686-musl doesn't really exist, so don't worry about it.
> 
> `[archname]` means glibc, `[archname]-musl` means musl, `[archname]-*` is both

I rewrote it as `x86_64-*` but it's now saying that the package can't be built for x86_64, any ideas as to why that is happening now?

**Comment by classabbyamp** [comment]
sorry `x86_64*` (no dash), but you shouldn't need to specify the archs, it should work on all of them

**Comment by sirkhancision** [comment]
> > Unfortunate to have a language server for bash being available only under `i686` & `x86_64`.
> 
> cross-compiles fine for me woman_shrugging

To me, it says:
`=> ERROR: nodejs-16.19.0_1: cannot be cross compiled...`

`=> ERROR: nodejs-16.19.0_1: host and target must have the same pointer size`

**Comment by classabbyamp** [comment]
it works on aarch64, but you shouldn't limit the arch because of that error: someone can still build it for themselves either natively or cross from 32bit

**Comment by sirkhancision** [comment]
> it works on aarch64, but you shouldn't limit the arch because of that error: someone can still build it for themselves either natively or cross from 32bit

Indeed, it built fine on aarch64. Thanks for the help with these packages, and the tips, you're awesome!

I think it might be fine, now.

**Comment by tjkirch** [comment]
Thanks @sirkhancision!  This is working great for me from a local build.  I also tested with bash-language-server 4.9.1 and it worked fine, in case you want to update the PR.

```diff
diff --git a/srcpkgs/bash-language-server/template b/srcpkgs/bash-language-server/template
index 02fcb44dfb..5c8f99cdd4 100644
--- a/srcpkgs/bash-language-server/template
+++ b/srcpkgs/bash-language-server/template
@@ -1,6 +1,6 @@
 # Template file for 'bash-language-server'
 pkgname=bash-language-server
-version=4.8.3
+version=4.9.1
 revision=1
 hostmakedepends="jq yarn"
 depends="nodejs"
@@ -9,7 +9,7 @@ maintainer="sirkhancision <jsantiago12tone@gmail.com>"
 license="MIT"
 homepage="https://github.com/bash-lsp/bash-language-server"
 distfiles="https://github.com/bash-lsp/bash-language-server/archive/refs/tags/server-${version}.tar.gz"
-checksum=a6c367e19a007b889f59fa7aa6a8c4fb0fae83fbfcc928e7a8f4366015dedbd6
+checksum=eef93c1167394fb01fc300c5b8c326fd9362dd3118a1e075549342e24e9d6e5c
 
 do_build() {
 	yarn
```

**Pull request merged**